### PR TITLE
Add TRIO106 rule

### PR DESCRIPTION
**Pull request by karpetrosyan**
## Summary

This PR is a part of #8451 and adds TRIO106 rule.

Examples

Bad
```python
import trio as asyncio
```

Bad
```python
from trio import move_on_after
````

Good
```python
import trio
```

### CODE REVIEW

**Comment by zanieb** [comment]
I'm not sure this limitation exists in Ruff. I believe we can detect the renamed import when determining if `trio` is being used.

**Comment by karpetrosyan** [comment]
> I'm not sure this limitation exists in Ruff. I believe we can detect the renamed import when determining if `trio` is being used.

I am not sure I understand what you mean here.

**Comment by charliermarsh** [comment]
@karpetrosyan - It seems like this rule exists in flake8-trio because flake8-trio needs `trio` to be imported as `trio` -- as in, `import trio` rather than `import trio as foo`.

In Ruff, we abstract that away. Whenever you use `resolve_call_path`, we resolve aliases. So if the user does `import trio as foo; foo.do_trio_thing()`, we'd understand that `foo.do_trio_thing()` is a reference to `trio.do_trio_thing`.

In other words, it seems like the limitation that exists in flake8-trio doesn't apply to Ruff, and so the rule wouldn't be needed. Does that make sense?

**Comment by karpetrosyan** [comment]
Yes, that makes perfect sense. Thanks @charliermarsh !

I am new to Ruff, so sorry if this PR was useless. Do I need to close it?

**Comment by charliermarsh** [comment]
@karpetrosyan - Not at all, no need to apologize! On the contrary, I'm sorry that you invested time in it. Yes, we can close the PR, since the rule doesn't seem relevant here.
